<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h2>Analyst: Margarette Felonia</h2>

In [1]:
import numpy as np
import pandas as pd

<h2>Converting Dataset(s) into Dataframe</h2>
<hr>

In [2]:
penguins = pd.read_csv("datasets//penguins.csv")

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Filling the null values</h1>
<hr>

In [9]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [10]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [11]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


<h1>Changing the Data type</h1>
<hr>

In [12]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [13]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h1>Creating your own column</h1>
<hr>

In [14]:
penguins['mass_kg'] = np.around(penguins['body_mass_g'] / 1000, 1)

In [15]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1>Explanatory Data Analysis(EDA)</h1>
<hr>

<h2>Question 1</h2>
<hr>
<h3>How many Chinstrap have a flipper lenght greater than 210g?</h3>

In [16]:
pgns_df = penguins[(penguins['species'] == 'Chinstrap') & (penguins['flipper_length_mm'] >= 210)]

In [17]:
pgns_df.value_counts().sum()

np.int64(4)

In [18]:
pgns_df 

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
313  Chinstrap  Dream            52.0           20.7              210.0   
316  Chinstrap  Dream            49.0           19.5              210.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  mass_kg  
313       4800.0  male  2008      4.8  
316       3950.0  male  2008      4.0  
323       4300.0  male  2009      4.3  
342       4100.0  male  2009      4.1

<h5>INSIGHT:   All 4 Chinstrap penguins with flipper length is greater than 210 mm are male and located on Dream Island. This suggests that within the Chinstrap species, males may develop longer flippers.</h5>

<h2>Question 2</h2>
<h3>How many penguins from Torgersen Island have a body mass of less than 3200 g?</h3>

In [19]:
pgns = penguins[(penguins['island'] == 'Torgersen') & (penguins['body_mass_g'] <= 3200)]

In [20]:
pgns.value_counts().sum()

np.int64(7)

In [21]:
pgns

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
12   Adelie  Torgersen            41.1           17.6              182.0   
68   Adelie  Torgersen            35.9           16.6              190.0   
80   Adelie  Torgersen            34.6           17.2              189.0   
116  Adelie  Torgersen            38.6           17.0              188.0   
120  Adelie  Torgersen            36.2           17.2              187.0   
124  Adelie  Torgersen            35.2           15.9              186.0   
128  Adelie  Torgersen            39.0           17.1              191.0   

     body_mass_g     sex  year  mass_kg  
12        3200.0  female  2007      3.2  
68        3050.0  female  2008      3.0  
80        3200.0  female  2008      3.2  
116       2900.0  female  2009      2.9  
120       3150.0  female  2009      3.2  
124       3050.0  female  2009      3.0  
128       3050.0  female  2009      3.0

<h5> INSIGHT: All 7 penguins with body mass less than 3200 g from Torgersen Island are female Adelie penguins. This indicates that Adelie females on Torgersen are typically smaller, and Torgersen may host one of the lightest penguin populations in the dataset.
</h5>

<h2>Question 3</h2>
<h3>How many penguins are recorded in the year 2008?</h3>

In [22]:
pings = penguins[penguins['year'] == 2008]

In [23]:
pings.value_counts().sum()

np.int64(114)

In [24]:
pings

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
50      Adelie  Biscoe            39.6           17.7              186.0   
51      Adelie  Biscoe            40.1           18.9              188.0   
52      Adelie  Biscoe            35.0           17.9              190.0   
53      Adelie  Biscoe            42.0           19.5              200.0   
54      Adelie  Biscoe            34.5           18.1              187.0   
..         ...     ...             ...            ...                ...   
315  Chinstrap   Dream            53.5           19.9              205.0   
316  Chinstrap   Dream            49.0           19.5              210.0   
317  Chinstrap   Dream            46.2           17.5              187.0   
318  Chinstrap   Dream            50.9           19.1              196.0   
319  Chinstrap   Dream            45.5           17.0              196.0   

     body_mass_g     sex  year  mass_kg  
50        3500.0  female  2008      3.5  
51        4300.0    male  2008      4.3  
52        3450.0  female  2008      3.4  
53        4050.0    male  2008      4.0  
54        2900.0  female  2008      2.9  
..           ...     ...   ...      ...  
315       4500.0    male  2008      4.5  
316       3950.0    male  2008      4.0  
317       3650.0  female  2008      3.6  
318       3550.0    male  2008      3.6  
319       3500.0  female  2008      3.5  

[114 rows x 9 columns]

<h5> INSIGHT: The year 2008 has a relatively large number of recorded penguins (114), indicating a high sampling activity or successful breeding season. This year could be good for conducting year-specific comparisons.</h5>

<h2>Question 4</h2>
<h3>How many penguins from Dream Island, recorded in the year 2008, have a bill length greater than or equal to 50 mm?</h3>

In [44]:
pengs = penguins[(penguins['island'] == 'Dream') & (penguins['year'] == 2008) & (penguins['bill_length_mm'] >= 50)]

In [45]:
pengs.value_counts().sum()

np.int64(7)

In [46]:
pengs

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
302  Chinstrap  Dream            50.5           18.4              200.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
318  Chinstrap  Dream            50.9           19.1              196.0   

     body_mass_g     sex  year  mass_kg  
302       3400.0  female  2008      3.4  
305       4550.0    male  2008      4.6  
307       4300.0    male  2008      4.3  
309       4100.0    male  2008      4.1  
313       4800.0    male  2008      4.8  
315       4500.0    male  2008      4.5  
318       3550.0    male  2008      3.6

<h5>INSIGHT: There were 7 penguins from Dream Island in 2008 with bill lengths greater than 50 mm, mostly Chinstrap males. This may indicate that larger-billed penguins are common in this region and year.</h5>

<h2>Question 5</h2>
<h3>How many female penguins have a body mass greater than or equal to 5000g?</h3>

In [28]:
pg = penguins[(penguins['sex'] == 'female') & (penguins['body_mass_g'] >= 5000)]

In [29]:
pg.value_counts().sum()

np.int64(8)

In [30]:
pg

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
173  Gentoo  Biscoe            45.1           14.5              215.0   
176  Gentoo  Biscoe            42.9           13.1              215.0   
184  Gentoo  Biscoe            45.1           14.5              207.0   
186  Gentoo  Biscoe            49.1           14.8              220.0   
200  Gentoo  Biscoe            44.9           13.3              213.0   
225  Gentoo  Biscoe            46.5           14.8              217.0   
262  Gentoo  Biscoe            50.5           15.2              216.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   

     body_mass_g     sex  year  mass_kg  
173       5000.0  female  2007      5.0  
176       5000.0  female  2007      5.0  
184       5050.0  female  2007      5.0  
186       5150.0  female  2008      5.2  
200       5100.0  female  2008      5.1  
225       5200.0  female  2008      5.2  
262       5000.0  female  2009      5.0  
274       5200.0  female  2009      5.2

<h5> INSIGHT: All 8 female penguins with body mass above 5000 g are Gentoo penguins from Biscoe Island. This reinforces that Gentoos are generally the heaviest species, particularly the females.
</h5>

<h2>Question 6</h2>
<h3>Which species of male penguins has the greatest bill depth?</h3>

In [31]:
penguins['bill_depth_mm'].max()

np.float64(21.5)

In [32]:
penguins[(penguins['sex'] == 'male') & (penguins['bill_depth_mm'] == 21.5)]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
19  Adelie  Torgersen            46.0           21.5              194.0   

    body_mass_g   sex  year  mass_kg  
19       4200.0  male  2007      4.2

<h5>INSIGHT:  
The greatest bill depth (21.5 mm) was found in a male Adelie penguin from Torgersen Island. While Adelies are not the largest species overall, some individuals show unique physical adaptations, such as greater bills.</h5>

<h2>Question 7</h2>
<h3>Which species has the highest recorded body mass (in kilograms) among all penguins?</h3>

In [55]:
penguins['mass_kg'].max()

np.float64(6.3)

In [61]:
penguins[penguins['mass_kg'] >= 6.3]['species']

169    Gentoo
Name: species, dtype: category
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [63]:
penguins[(penguins['species'] == 'Gentoo') & (penguins['mass_kg'] == 6.3)]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  mass_kg  
169       6300.0  male  2007      6.3

<h5> INSIGHT: The heaviest penguin recorded was a male Gentoo weighing 6.3 kg, found on Biscoe Island in 2007—showing that Gentoo penguins are the largest among the species in the dataset.</h5>